tests/labeler.test.ts

```diff
@@ -0,0 +1,84 @@
+jest.mock('danger', () => jest.fn());
+import * as danger from 'danger';
+const dm = danger as any;
+
+import { labeler } from '../org/labeler';
+
+beforeEach(() => {
+  dm.danger = {
+    github: {
+      repository: {
+        name: 'gatsby',
+        owner: {
+          login: "gatsbyjs"
+        }
+      },
+      issue: {
+        labels: [],
+        number: 100
+      },
+      api: {
+        issues: {
+          addLabels: jest.fn()
+        }
+      }
+    }
+  };
+});
+
+describe('a new issue', () => {
+  it('with question mark in a title', () => {
+    dm.danger.github.issue.title = 'Help - Has anyone hosted a gatsby.js site on Platform.sh?';
+    return labeler().then(() => {
+      expect(dm.danger.github.api.issues.addLabels).toBeCalledWith({
+        repo: 'gatsby',
+        owner: 'gatsbyjs',
+        number: 100,
+        labels: ['type: question or discussion']
+      })
+    });
+  });
+
+  it('starting with how', () => {
+    dm.danger.github.issue.title = 'How do you justify Gatsby’s bundle size to clients';
+    return labeler().then(() => {
+      expect(dm.danger.github.api.issues.addLabels).toBeCalledWith({
+        repo: 'gatsby',
+        owner: 'gatsbyjs',
+        number: 100,
+        labels: ['type: question or discussion']
+      })
+    });
+  });
+
+  it('including tutorial', () => {
+    dm.danger.github.issue.title = 'Tutorial template + gold standard example';
+    return labeler().then(() => {
+      expect(dm.danger.github.api.issues.addLabels).toBeCalledWith({
+        repo: 'gatsby',
+        owner: 'gatsbyjs',
+        number: 100,
+        labels: ['type: documentation']
+      })
+    });
+  });
+
+  it('including readme', () => {
+    dm.danger.github.issue.title = '[v2] default starter: update README';
+    return labeler().then(() => {
+      expect(dm.danger.github.api.issues.addLabels).toBeCalledWith({
+        repo: 'gatsby',
+        owner: 'gatsbyjs',
+        number: 100,
+        labels: ['type: documentation']
+      })
+    });
+  });
+
+  it('not recognised', () => {
+    dm.danger.github.issue.title = 'Supporting HSTS and how to HSTS preloading';
+    return labeler().then(() => {
+      expect(dm.danger.github.api.issues.addLabels).not.toBeCalled()
+    })
+  });
```

Comment:
Shall we leave it as is till we have a discussion and update it with next PR or do you want to wait?

Reply:
Yeah let's go with unlabelled to start with, then we can look into adding a specific "this needs labelling" label if that doesn't work out.

Reply:
We can filter by "Unlabeled" ( https://github.com/gatsbyjs/gatsby/issues?q=is%3Aopen+is%3Aissue+no%3Alabel ), so maybe new label is not needed? 

Could this run one time against existing unlabeled issues to try to label them too?

Reply:
In theory we could play around with writing a task to do periodic check on issues without labels. This could be useful when new parser logic is introduced to handle more label types.

Reply:
> In theory we could play around with writing a task to do periodic check on issues without labels. This could be useful when new parser logic is introduced to handle more label types.

Trying to think of a good way to trigger a task like that:
- Peril watches for a comment event on _this_ repo
- Comment must contain `<trigger phrase>`
- Comment must be from user that is part of the org (or other higher permissions?)

Reply:
I thought about using https://github.com/danger/peril/blob/master/docs/settings_repo_info.md#tasks

Reply:
I think the scheduler might make more sense, maybe?

See [these settings](https://github.com/artsy/peril-settings/blob/master/peril.settings.json#L35-L37) and [this task](https://github.com/artsy/peril-settings/blob/master/tasks/daily-license-check.ts) for what I mean. It looks like we can set a task to run at set intervals that will do pretty much whatever we want it to, and that could allow us to:

- Search for [issues with no labels](https://github.com/gatsbyjs/gatsby/issues?q=is%3Aopen+is%3Aissue+no%3Alabel+sort%3Aupdated-desc)
- Check if they’re label-able now
- Label them if so

Reply:
Yup, scheduler makes more sense for sure. I just forgot that Heroku offers a free 500mb MongoDB plan so you can use the scheduler with that 👌